### ISSUE TITLE
cargo clippy --fix doesn't remove dead code

### ISSUE THREAD

**Issue by jrmuizel** [rust-lang/rust-clippy#8713]
### Description

It would be nice if `cargo clippy --fix` could remove the code that it complains about being dead.

### Version

```text
rustc 1.59.0 (9d1b2106e 2022-02-23)
```

**Comment by xFrednet** [comment]
Hey, could you provide an example for this? The normal `dead_code` comes to mind, which is implemented in rustc. :upside_down_face:

**Comment by Oliboy50** [comment]
`cargo fix` should already remove the dead code, no?

**Comment by xFrednet** [comment]
The effect of `cargo fix` is defined by the [applicability](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint_defs/enum.Applicability.html#variants) of the suggestion. I believe that `dead_code` doesn't actually suggest removing the code. It's mostly a lint to notify the user, in most cases users want to keep the original code. :)

If you really believe this would be a good addition, you should raise this issue in the `rust-lang/rust` repo. Though, I think the current behavior is intentional and good.
